### ISSUE TITLE
afterLoad always fires first index=1 even accessing the page on anchor=3

### ISSUE THREAD

**Issue by medfx** [alvarotrigo/fullPage.js#2519]
Hi, I have a question about the afterLoad callback:

When I open a deep link to a fullpage.js site (e.g. http://alvarotrigo.com/fullPage/#3rdPage) the afterLoad callback get once called with index = 1 then the page scrolls to the third section and afterLoad will be called again with index = 3. So no matter where I really wanna go, the first section will always load fully once?

I'm asking about this behavior because I'm lazy loading heavy image backgrounds with the help of the afterLoad callback. But if the user doesn't enter the page on the first section, the images load anyway because afterLoad always fires on the first section first.

How can I determine if a user is just passing through the first section on his way to another section first, or if he is loading the first section to stay on it?

I would provide a jsfiddle but I dont know how to set a deep link into the jsfiddle result frame.

**Comment by alvarotrigo** [comment]
I can't reproduce your issue: 
http://jsfiddle.net/97tbk/1658/

Closed until an isolated reproduction is provided as detailed in the [reporting issues guidelines](https://github.com/alvarotrigo/fullPage.js#reporting-issues) and the [issue skeleton](https://github.com/alvarotrigo/fullPage.js/blob/master/.github/ISSUE_TEMPLATE.md).

**Comment by medfx** [comment]
It does not work if you do it with the "active" class. The user has to to come through a deep link.

I found out how to open the jsfiddle iframe directly and send anchor links:
Result frame: http://fiddle.jshell.net/97tbk/1659/show/#section3
Jsfiddle: http://jsfiddle.net/97tbk/1666/

**Comment by alvarotrigo** [comment]
Oh, you mean by accessing through the anchors. I see. 
I'll open it again and research it a bit more.

**Comment by medfx** [comment]
I also tested it with the CSS .active class. When you go directly to e.g. the third section (#thirdSection) fullpage.js toggles the active class on the home/first section for a very shot time.

Also the images implemented with lazy-loading ("data-src") load on the home/first section, when you enter the page on any other section.

**Comment by alvarotrigo** [comment]
> I also tested it with the CSS .active class. When you enter directly e.g. the third section (#thirdSection) fullpage.js toggles the active class on the home/first section for a very short time.

I don't think that's the case. You are just probably seeing the 1st section as the library needs to apply its css and classes. If you do it on the header you shouldn't get the blink. 
If you think that's the case, then you'll have to provide some prove of it regarding the addition of the `active` class.

**Comment by medfx** [comment]
You have to come trough a anchor link. It does not work if the `active` class is already assigned to a section.

I uploaded the files onto a normal webserver so we don't have the problem with the changing url of the codepen/jsfiddle iframe and can attach anchor links.

http://www.arisur.info/fullpage/index.html#Section3
So we enter here directly trough the anchor `#Section3`. None of the sections has the active class preassigned.
Console says:
```
afterLoad: sectionIndex 1
section-1 has active class
afterLoad: sectionIndex 3
```

Also the image on section 1 which has no `src` only `data-src` for lazy loading will load, if we enter the site directly on `#Section3`. (You can check this in developer tools network tab).

If we set the option `animateAnchor: false` the site will directly load section 3. No blinking no jumping, but `afterLoad` will still trigger on section 1, the active class will be toggled and the image on section 1 loads.

**Comment by alvarotrigo** [comment]
> Edit: Oh now I understand where I was unclear. I tested the automatic toggle of the active class. I didn't ment I tested this behavior with the preassigned active class.

Yeah right. Then we agree :) 
I'll investigate the issue regarding the URL anchor link and the after load whenever I have time. I'll mark it as a bug.

**Comment by alvarotrigo** [comment]
Fixed on the dev branch 2.9.4

**Comment by alvarotrigo** [comment]
Fixed in the latest release 2.9.4
